### PR TITLE
GEOMESA-2815 Add option to disable coprocessors

### PR DESCRIPTION
**Pull request by elahrvivaz**
* New HBase datastore params:
  * `hbase.coprocessor.arrow.enable` default false
  * `hbase.coprocessor.bin.enable` default true
  * `hbase.coprocessor.density.enable` default true
  * `hbase.coprocessor.stats.enable` default true
* Standardize data store config classes

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
stacks on top of #2504

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala [hunk 1]
```diff
@@ -216,4 +213,34 @@ object AccumuloDataStoreFactory extends GeoMesaDataStoreInfo {
...
+      authProvider: AuthorizationsProvider,
+      audit: Option[(AuditWriter with AuditReader, AuditProvider, String)],
+      queries: AccumuloQueryConfig,
+      writeThreads: Int,
+      namespace: Option[String]
+    ) extends GeoMesaDataStoreConfig
+
+  case class AccumuloQueryConfig(
+      threads: Int,
+      recordThreads: Int,
```
Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala]: What's the difference between 'threads' and 'recordThreads'?

File: geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreFactory.scala [hunk 2]
```diff
@@ -36,39 +36,48 @@ class HBaseDataStoreFactory extends DataStoreFactorySpi with LazyLogging {
...
+      looseBBox = LooseBBoxParam.lookup(params),
+      caching = CachingParam.lookup(params),
+      maxRangesPerExtendedScan = MaxRangesPerExtendedScanParam.lookup(params)
+    )
+    val enabledCoprocessors = EnabledCoprocessors(
+      arrow = ArrowCoprocessorParam.lookup(params),
+      bin = BinCoprocessorParam.lookup(params),
+      density = DensityCoprocessorParam.lookup(params),
+      stats = StatsCoprocessorParam.lookup(params)
+    )
```
Comment (jnh5y) [inline geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreFactory.scala]: Should we have a check to see if someone turned off remote filters and yet did something else with enabledCoprocessors?

File: geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreParams.scala [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      "Additional HBase configuration properties, as a standard XML `<configuration>` element",
+      largeText = true,
+      supportsNiFiExpressions = true)
+
+  val ArrowCoprocessorParam =
+    new GeoMesaParam[java.lang.Boolean](
+      "hbase.coprocessor.arrow.enable",
+      "Processes Arrow encoding in HBase region servers as a coprocessor call",
+      default = Boolean.box(false),
+      supportsNiFiExpressions = true
```
Comment (jnh5y) [inline geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreParams.scala]: Ok, so as a question, do we want that these options are available in NiFi?

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala (hunk 1 above)
Comment (elahrvivaz) [reply geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala]: recordThreads is for queries against the records/`id` table - we've always had that as a separate config in accumulo although nowhere else...

File: geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreParams.scala (hunk 3 above)
Comment (elahrvivaz) [reply geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreParams.scala]: they will get added automatically i believe, based on being in the data store factory param list.

File: geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreFactory.scala (hunk 2 above)
Comment (elahrvivaz) [reply geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreFactory.scala]: mmm, we could... do you think it's confusing?

Comment (jnh5y) [reply geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreFactory.scala]: I was there for both features, so I'm clear on what's happening:).  I have no idea what other people's reaction will be.

**Comment by jnh5y** [comment]
merged as 580b190da4f177018f3ea2944c15e428cbf2978b

**Pull request closed without merging**